tests/test_smoketest.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+"""
+A basic integration test of the DSS. This can also be invoked via `make smoketest`.
+"""
+
+import os, sys, argparse, time, uuid, json, shutil, tempfile, unittest
+from subprocess import check_call, check_output, CalledProcessError
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss.api.files import ASYNC_COPY_THRESHOLD
+from tests import infra
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--no-clean', dest='clean', action='store_false',
+                    help="Don't remove the temporary working directory on exit.")
+args = argparse.Namespace(clean=True)
+
+def GREEN(message=None):
+    if message is None:
+        return "\033[32m" if sys.stdout.isatty() else ""
+    else:
+        return GREEN() + message + ENDC()
+
+def RED(message=None):
+    if message is None:
+        return "\033[31m" if sys.stdout.isatty() else ""
+    else:
+        return RED() + message + ENDC()
+
+def ENDC():
+    return "\033[0m" if sys.stdout.isatty() else ""
+
+def run(command, runner=check_call, **kwargs):
+    if isinstance(command, str):
+        kwargs["shell"] = True
+    print(GREEN(command))
+    try:
+        return runner(command, **kwargs)
+    except CalledProcessError as e:
+        parser.exit(RED(f'{parser.prog}: Exit status {e.returncode} while running "{command}". Stopping.'))
+
+
+@infra.testmode.standalone
```

Comment:
I didn't know you could the decorator works on the whole class. That's handy. Also is this an integration test?

Reply:
D'oh. Corrected.